ARMV5 can use STRD and LDRD access instructions but these accesses need to be 
8 byte aligned. The dynamic linker's malloc needs to match this so structures
become 8 byte aligned to void unaligned accesses.

RP - 14/02/2008

Index: uClibc-0.9.29/ldso/ldso/arm/dl-sysdep.h
===================================================================
--- uClibc-0.9.29.orig/ldso/ldso/arm/dl-sysdep.h	2008-02-14 00:58:12.000000000 +0000
+++ uClibc-0.9.29/ldso/ldso/arm/dl-sysdep.h	2008-02-14 00:59:19.000000000 +0000
@@ -15,6 +15,8 @@
   GOT_BASE[1] = (unsigned long) MODULE; \
 }
 
+#define DL_MALLOC_ALIGN 8  /* EABI needs 8 byte alignment for STRD LDRD*/
+
 static inline unsigned long arm_modulus(unsigned long m, unsigned long p)
 {
 	unsigned long i,t,inc;
